website/static/js/share/utils.js

```diff
@@ -87,6 +111,28 @@ var maybeQuashEvent = function(event) {
     }
 };
 
+var addFilter = function(vm, filter, required){
+    required = required || false;
+    if (required === true && vm.requiredFilters.indexOf(filter) === -1){
+        vm.requiredFilters.push(filter);
+    } else if (vm.optionalFilters.indexOf(filter) === -1){
+        vm.optionalFilters.push(filter);
+    }
+    search(vm);
```

Comment:
Maybe rename the function to `updateFilters`? 
`addFilter` doesn't strike me as something that would have side effects.